__tests__/components/ActiveLink.test.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import * as nextRouter from 'next/router';
+import React from 'react';
+import { cleanup, render, waitFor, screen } from '@testing-library/react';
+import ActiveLink from 'components/ActiveLink';
+
+describe('ActiveLink Component', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    nextRouter.useRouter = jest.fn();
+  });
+
+  afterEach(cleanup);
+
+  it('provides non-active link', async () => {
+    nextRouter.useRouter.mockImplementation(() => ({
+      route: '/',
+      asPath: '/',
+      prefetch: jest.fn(() => Promise.resolve()),
+    }));
+    render(
+      <ActiveLink href="/whatever/" activeClassName="active">
+        <a>test-link</a>
+      </ActiveLink>,
+    );
+    await waitFor(() => screen.getByRole('link'));
+    const link = screen.getByRole('link');
```

Comment:
Can you combine this with the `waitFor` above? Something like:

`const link = await waitFor(() => screen.getByRole('link'));`